conan/tools/gnu/gnudeps_flags.py

```diff
@@ -80,6 +80,11 @@ def _format_libraries(self, libraries):
                 if not library.endswith(".lib"):
                     library += ".lib"
                 result.append(library)
+            elif library.startswith("lib") and library.endswith(".a"):
+                # link by filename
+                # TODO: at this point, we should only use -l:
+                # if we are sure that we use ld as linker
```

Comment:
Probably the way to go, aligned with Conan 2.0 design, is not try to detect it, but use a ``conf`` that explicitly defines it (opt-in or opt-out depending on the default we chose). Better than tricky, and not 100% robust auto-detections.